src/date/index.ts

```diff
@@ -27,3 +31,344 @@ export function getMonthAbbrByIndex(monthNumber: number): string {
+ * getPreviousMonthDate('1992-10-10');  // Output will be 'Thu Sep 10 1992 05:30:00 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate(new Date());  // Output will be 'Tue Jan 04 2022 02:28:43 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate('03/25/2015');  // Output will be 'Wed Feb 25 2015 00:00:00 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate();  // Output will be 'Invalid date'
+ * getPreviousMonthDate('03/31/2022');  // Output will be 'Thu Mar 03 2022 00:00:00 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate(new Date(), 10); // Output will be 'Sun Apr 04 2021 15:18:48 GMT+0530 (India Standard Time)'
+ * ```
+ */
+export function getPreviousMonthDate(date: Date, monthCount: number = 1) {
+  try {
+    const newDate = new Date(date);
+
+    newDate.setMonth(newDate.getMonth() - monthCount);
+
+    return newDate;
+
+  } catch (error) {
+    console.error('Error in getPreviousMonthDate: ', error);
+  }
+}
+
+
+/**
+ * This method is used for getting previous date to the date provided through argument. argument should come through
+ * only new Date()
+ * Not doing with dayjs as no specific methods for this use case are there.
+ *
+ * @param {Date} date - Date as only new Date()
+ * @param {number} date - number of days we want to go back to
+ *
+ * @example
+ * ```
+ * getPreviousDayDate(new Date());  // Output will be 'Thu Feb 03 2022 03:28:49 GMT+0530 (India Standard Time)'
+ * ```
+ */
+export function getPreviousDayDate(date: Date, daysCount: number = 1) {
+  try {
+    const newDate = new Date(date);
+
+    const newFinalDate = new Date(newDate.getTime() - (1000 * 60 * 60 * 24 * daysCount));
+
+    return newFinalDate;
+
+  } catch (error) {
+    console.error('Error in getPreviousDayDate: ', error);
+  }
+}
+
+
+/**
+ * This function converts DD/MM/YYYY to YYYY-MM-DD.
+ * JavaScript new Date() method accept date in YYYY-MM-DD or YYYY/MM/DD
+ * format that's why we needed this method
+ * This function is needed because dayjs does not work with 'DD/MM/YYYY' type of inputs
+ *
+ * @param {string} date - Input date string in DD/MM/YYYY or DD-MM-YYYY format
+ *
+ * @example
+ * ```
+ * convertStrToValidDateFormat('22/11/2021');  - // Output will be '2021-11-22'
+ * convertStrToValidDateFormat('03/25/2015');  - // Output will be '2015-25-03'
+ * convertStrToValidDateFormat('03-25-2015');  - // Output will be '2015-25-03'
+ * ```
+ */
+export function convertStrToValidDateFormat(date: string) {
+  try {
+    const str = date.replace(/[^0-9]/g, '');
+
+    if (str.length !== 8) {
+      throw new Error('Error in converting date\'s format');
+    }
+
+    const result = str.slice(4, 8) + '-' + str.slice(2, 4) + '-' + str.slice(0, 2);
+
+    return result;
+
+  } catch (err) {
+    console.error('Error in converting date\'s format', err);
+    return '';
+  }
+}
+
+
+/**
+ * This function checks if input date is valid or not.
```

Comment:
can you give context what you mean by valid date? Like 100 years old dates are not valid or 32 Jan ... what you mean by valid date? Unable to understanf even from examples.

Reply:
updated